technique/muon/zero_field_setup.py

```diff
@@ -0,0 +1,278 @@
+        :return: instrument name
+        """
+        return g.get_instrument()
+
+
+def update_offset():
+    """
+    updates the offset so that the corrected field reading is 0 for each axis
+    :return: None
+    """
+    x_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:X:_RAW")
+    y_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Y:_RAW")
+    z_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Z:_RAW")
+
+    x_new_offset = x_raw_measurement - 0
+    y_new_offset = y_raw_measurement - 0
+    z_new_offset = z_raw_measurement - 0
+
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:X:OFFSET", x_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Y:OFFSET", y_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Z:OFFSET", z_new_offset)
+
+
+def run_all():
+    """
+    Run all the functions in order for set up
+    :return:
+    """
+    while True:
+        print("***********************")
+        procedure = ZeroFieldSetupProcedure()
+        procedure.set_setpoints_to_zero(0, 0, 0)
+        # 2 second timer
+        if procedure.check_if_stray_field_exist():
+            print("Warning! Magnet Not in Range")
+            # pylint: disable=E0602
+
+        current_x, current_y, current_z, fields_x, fields_y, fields_z = procedure.get_correlated_current_and_fields(
+            plot=True)
+
+        calibration_coefficient_x, coefficient_of_determination_x = procedure.calculate_coefficient_and_r_squared(
+            current_x, fields_x)
+        calibration_coefficient_y, coefficient_of_determination_y = procedure.calculate_coefficient_and_r_squared(
+            current_y, fields_y)
+        calibration_coefficient_z, coefficient_of_determination_z = procedure.calculate_coefficient_and_r_squared(
+            current_z, fields_z)
+
+        print(
+            "Calibration coefficient for X axis is {unknown}".format(
+                unknown=calibration_coefficient_z))
+        print("The coefficient of determination is {unknown}\n".format(
+            unknown=coefficient_of_determination_x))
+
+        print(
+            "Calibration coefficient for Y axis is {unknown}".format(
+                unknown=calibration_coefficient_y))
+        print("The coefficient of determination is {unknown}\n".format(
+            unknown=coefficient_of_determination_y))
+
+        print(
+            "Calibration coefficient for Z axis is {unknown}".format(
+                unknown=calibration_coefficient_z))
+        print("The coefficient of determination is {unknown}\n".format(
+            unknown=coefficient_of_determination_z))
+
+        if procedure.ask_user_to_continue() == "N":
+            break
+
+        RMS = procedure.calculate_noise(plot=True)
+        print("RMS noise: {unknown}".format(unknown=RMS))
+
+        if procedure.ask_user_to_continue() == "N":
+            break
+
+        update_offset()
+        procedure.close_plot()
+        break
+
+    procedure.close_plot()
+    print("Script Stopped")
```

Comment:
We can probably leave it up to the user to close the plot and it should be obvious when the script is done without printing.